docs/architecture/ingest.md

```diff
@@ -0,0 +1,87 @@
+# Ingest
+
+This is a component that is responsible for ingesting and processing data.
+
+It exposes a GRPC API for Osmosis node clients to send data to be ingested.
+
+See [this ADR](https://www.notion.so/osmosiszone/ADR-006-Streaming-GRPC-SQS-Ingest-4e3b2ff7d23e43e2a1f3c43adc3c26bc) for details about the design.
+
+The node pushes only the pool data that is changed within a block.
+
+As a result, it is possible to see the following sequence of events:
+- Height X: All Osmosis pools are pushed
+- Height X+1: Only the pools that have changed within that height are pushed
+
+## Workers
+
+### Pricing
+
+The pricing worker is responsible for pre-computing prices for all assets and the default quote denom (USDC).
+
+It is triggerred by the ingester asyncronously when the GRPC push is received.
+
+It looks at the unique denoms constructed from the pool data and computes the prices for each
```

Comment:
Want to set the right context for myself before diving deeper in - for pools having default denom, i.e. USDC or USDT, this is self-explained. What about non-default denom pools, e.g. ATOM-OSMO. 
Please ignore if this is explained later.

Reply:
It computes the prices with the default quote denom (configured to USDC via config)

Extended this for clarity

Reply:
Actually, it already says that right above

> The pricing worker is responsible for pre-computing prices for all assets and the default quote denom (USDC)